pipeline_steps/common.groovy

```diff
@@ -1240,19 +1240,19 @@ void internal_slave(Closure body){
 }
 
 void standard_job_slave(String slave_type, Closure body){
-  timeout(time: 8, unit: 'HOURS'){
+  timeout(time: env.JOB_TIMEOUT_HR, unit: 'HOURS'){
```

Comment:
This isn't really a job timeout, this is a slave timeout. A job may use multiple slaves, in parallel or serial, and its not unreasonable for the job as a whole to take longer than any one slave is used for. 

I guess what I'm saying is that if these are going to be tweak-able they should probably be tweak-able independently.

Both should have a default, so most users will only need to set `JOB_TIMEOUT` but `AGENT_TIMEOUT` could be useful in some situations.

Reply:
Refactored to accomodate additional parameter.